python/dgl/sparse/elementwise_op_sp.py

```diff
@@ -89,14 +81,15 @@ def sp_mul(
     values=tensor([2, 4, 6]),
     shape=(3, 4), nnz=3)
     """
-    if isinstance(A, SparseMatrix) and isinstance(B, (float, int)):
+    if isinstance(B, (float, int)):
         return val_like(A, A.val * B)
-    elif isinstance(A, (float, int)) and isinstance(B, SparseMatrix):
+    elif isinstance(B, SparseMatrix):
```

Comment:
What if both `A` and `B` are SparseMatrix objects? I got the error message "TypeError: unsupported operand type(s) for *: 'SparseMatrix' and 'Tensor'", which can be confusing.

Reply:
In general one should raise an "unsupported operand" TypeError if the two types are incompatible for some operation.
```
>>> torch.tensor([2,3,4]) / 'aaa'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for /: 'Tensor' and 'str'
```